src/Symfony/Component/Translation/Command/TranslationLintCommand.php

```diff
@@ -0,0 +1,129 @@
+ *
+ * @author Hugo Alliaume <hugo@alliau.me>
+ */
+#[AsCommand(name: 'lint:translations', description: 'Lint translations files syntax and outputs encountered errors')]
+class TranslationLintCommand extends Command
+{
+    private SymfonyStyle $io;
+
+    public function __construct(
+        private TranslatorInterface&TranslatorBagInterface $translator,
+        private array $enabledLocales = [],
+    ) {
+        parent::__construct();
+    }
+
+    public function complete(CompletionInput $input, CompletionSuggestions $suggestions): void
+    {
+        if ($input->mustSuggestOptionValuesFor('locales')) {
+            $suggestions->suggestValues($this->enabledLocales);
+        }
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setDefinition([
+                new InputOption('locales', null, InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY, 'Specify the locales to lint.', $this->enabledLocales),
+            ])
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</> command lint translations.
+
+  <info>php %command.full_name%</>
+EOF
+            );
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->io = new SymfonyStyle($input, $output);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $locales = $input->getOption('locales');
+
+        /** @var array<string, array<string, array<string, \Throwable>> $errors */
+        $errors = [];
+        $domainsByLocales = [];
+
+        foreach ($locales as $locale) {
+            $messageCatalogue = $this->translator->getCatalogue($locale);
+
+            foreach ($domainsByLocales[$locale] = $messageCatalogue->getDomains() as $domain) {
+                foreach ($messageCatalogue->all($domain) as $id => $translation) {
+                    try {
+                        $this->translator->trans($id, [], $domain, $messageCatalogue->getLocale());
+                    } catch (ExceptionInterface $e) {
+                        $errors[$locale][$domain][$id] = $e;
+                    }
+                }
+            }
+        }
+
+        if ([] === $domainsByLocales) {
+            $this->io->error('No translation files were found.');
+
+            return Command::FAILURE;
+        }
+
+        $this->io->table(
+            ['Locale', 'Domains', 'Valid?'],
+            array_map(
+                static fn (string $locale, array $domains) => [
+                    $locale,
+                    implode(', ', $domains),
+                    empty($errors[$locale]) ? '<info>Yes</>' : '<error>No</>',
+                ],
+                array_keys($domainsByLocales),
+                $domainsByLocales
+            ),
+        );
+
+        if ([] !== $errors) {
+            foreach ($errors as $locale => $domains) {
+                foreach ($domains as $domain => $errors) {
```

Comment:
```suggestion
                foreach ($domains as $domain => $domainErrors) {
```
To avoid overriding `$errors` variable.